### PR TITLE
Spec adAuctionComponents

### PR DESCRIPTION
**Pull request by HabibiYou**
Addresses https://github.com/WICG/turtledove/issues/1174

***
<a href="https://pr-preview.s3.amazonaws.com/HabibiYou/fenced-frame/pull/163.html" title="Last updated on Jun 4, 2024, 2:56 PM UTC (162281a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/fenced-frame/163/e67a4f9...HabibiYou:162281a.html" title="Last updated on Jun 4, 2024, 2:56 PM UTC (162281a)">Diff</a>

### CODE REVIEW

File: spec.bs [hunk 1]
```diff
@@ -1505,6 +1506,39 @@ partial interface Navigator {
...
+    origin=], then then throw a {{DOMException}}.
+
+  1. Let |maxAdComponents| be 40.
+
+  1. If |numAdComponents| is greater than |maxAdComponents|, then set |numAdComponents| to 
+    |maxAdComponents|.
+
+  1. Let |adComponentsURNs| be an empty [=list=] of [=fencedframeconfig/urn=]s.
+
+  1. [=map/For each=] |urn| → _ of |instance|'s [=fenced frame config instance/nested
```
Comment (domfarolino) [inline spec.bs:1529]: We don't really use `_` in specs I don't think? Just make this an ignored variable I think.

File: spec.bs [hunk 2]
```diff
@@ -1505,6 +1506,39 @@ partial interface Navigator {
...
+
+  1. If |instance| is null, then throw a {{DOMException}}.
+
+  1. If [=this=]'s [=relevant settings object=]'s [=environment settings object/origin=] and
+    |instance|'s [=fenced frame config instance/mapped url=]'s [=url/origin=] are not [=same
+    origin=], then then throw a {{DOMException}}.
+
+  1. Let |maxAdComponents| be 40.
+
+  1. If |numAdComponents| is greater than |maxAdComponents|, then set |numAdComponents| to
```
Comment (domfarolino) [inline spec.bs:1524]: I would just use the `&gt;` sign here directly. Will probably need to adjust the wrapping after.

File: spec.bs [hunk 3]
```diff
@@ -1505,6 +1506,39 @@ partial interface Navigator {
...
+  1. Let |adComponentsURNs| be an empty [=list=] of [=fencedframeconfig/urn=]s.
+
+  1. [=map/For each=] |urn| → _ of |instance|'s [=fenced frame config instance/nested
+     configs=]:
+     
+     1. If |numAdComponents| equals 0, then [=iteration/break=].
+
+     1. [=list/Append=] |urn| to |adComponentsURNs|.
+
+     1. Decrement |numAdComponents| by 1.
```
Comment (domfarolino) [inline spec.bs:1536]:
```suggestion
     1. Set |numAdComponents| to |numAdComponents| &minus; 1.
```

File: spec.bs (hunk 1 above)
Comment (HabibiYou) [reply spec.bs:1529]: Sounds good to me.

**Review by domfarolino — COMMENTED** [review]
Same question as https://github.com/WICG/fenced-frame/pull/161#pullrequestreview-2097119348

**Comment by JensenPaul** [comment]
> Same question as [#161 (review)](https://github.com/WICG/fenced-frame/pull/161#pullrequestreview-2097119348)

I feel like `adAuctionComponents()` belongs here as it references only terms defined in the FF spec.  Were it in the PA spec, it would have to cross-reference everything from the FF spec, and would hence be a bit non sequitur and lack relevant context for readers.

**Review by domfarolino — APPROVED** [review]

**Pull request merged**